test/test_queue_routes.py

```diff
@@ -3,83 +3,360 @@
 
 
-def test_get_queue(client):
+def test_queue_get(client):
     """Test if we can get the queue."""
-
     rv = client.get('/mxcube/api/v0.1/queue')
     assert rv.status_code == 200
 
 
-def test_get_item(client):
+def test_queue_get_item(client):
     """Test if we the queue has the intial sample, only mockups hwobj."""
     rv = client.get('/mxcube/api/v0.1/queue')
-
     assert (rv.status_code == 200 and json.loads(rv.data).get('1:01'))
 
-
-def test_add_sample(client):
+def test_queue_add_item(client):
     """Test if we can add a sample."""
-    sample_to_add = {
-        'code': 'matr1_5',
-        'checked': True,
-        'sampleName': 'Sample-105',
-        'sampleID': '1:05',
-        'tasks': [],
-        'location': '1:5',
-        'defaultPrefix': 'local-user',
-        'type': 'Sample'
-    }
+    rv = client.post('/mxcube/api/v0.1/queue',
+                     data=json.dumps([test_sample]),
+                     content_type='application/json'
+                     )
+    assert rv.status_code == 200
+
+def test_add_and_get_sample(client, add_sample):
+    """Test if we can add a sample. The sample is added by a fixture."""
+    rv = client.get('/mxcube/api/v0.1/queue')
+    assert (rv.status_code == 200 and json.loads(rv.data).get('1:05'))
+
+def test_add_and_get_task(client, add_sample, add_task):
+    """Test if we can add a task to the sample."""
+    rv = client.get('/mxcube/api/v0.1/queue')
+    assert (rv.status_code == 200 and len(json.loads(rv.data).get('1:05')['tasks']) == 1)
+
+def test_add_and_edit_task(client, add_sample, add_task):
+    """Test if we can add edit a task i the sample in the queue."""
+    rv = client.get('/mxcube/api/v0.1/queue')
+    queue_id = json.loads(rv.data).get('1:05')['queueID']
+    task_queue_id = json.loads(rv.data).get('1:05')['tasks'][0]['queueID']
+    assert (rv.status_code == 200 and len(json.loads(rv.data).get('1:05')['tasks']) == 1)
+
+    task_to_update = copy.deepcopy(test_edit_task)
+    parameter_to_update = 'num_images'
+    parameter_update_value = 10
+    task_to_update['parameters'][parameter_to_update] = parameter_update_value
+    rv = client.post(('/mxcube/api/v0.1/queue/{}/{}').format(queue_id, task_queue_id),
+                     data=json.dumps(task_to_update),
+                     content_type='application/json'
+                     )
+    assert (rv.status_code == 200 and json.loads(rv.data).get('parameters')[parameter_to_update] == parameter_update_value)
+
+def test_queue_start(client, add_sample, add_task):
+    """Test if we can start the queue. The queue requires a sample and a task to start which are added by fixtures."""
+    rv = client.put('/mxcube/api/v0.1/queue/start',
+                    data=json.dumps({'sid':'1:05'}),
+                    content_type='application/json')
+    assert rv.status_code == 200
+
+def test_queue_stop(client, add_sample, add_task):
+    """Test if we can stop the queue. The queue is started and then stopped."""
```

Comment:
Same here, we could also check if the queue was actually stopped ?

Reply:
Same problem as with the start.